Photos/app/src/main/java/hieunguyen/com/photos/HttpHandler.java

```diff
@@ -0,0 +1,93 @@
+package hieunguyen.com.photos;
+
+import android.content.Context;
+import android.net.Uri;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+/**
+ * Created by hieunguyen on 9/17/15.
+ * HTTP request and response manager
+ */
+public class HttpHandler {
+
+    public RequestQueue queue;
+    public Context context;
+
+    public HttpHandler(Context context) {
+        this.context = context;
+        queue = Volley.newRequestQueue(context);
+    }
+
+    /**
+     * Uses Google API search and passes the results to the callback function
+     * @param searchQuery What the user wants to look for. Parameter 'q' of URL
+     * @param startIndex Helps paging in Search Fragment and avoids repeated results
+     * @param callback Callback function to pass the processed returned JSON to the caller
+     */
+    public void performSearch(String searchQuery, int startIndex, final SuccessCallback callback) {
+        String query = searchQuery.replaceAll(" ", "+");
```

Comment:
this is a super long method, it should be refactored into a couple smaller methods. rule of thumb: you should be able to see the entire method on your screen (declaration to final } ).